evm/router/router.go

```diff
@@ -228,30 +223,22 @@ func (r *routerImpl) getConnectionForNamespace(ns string) (routerv1.MsgClient, e
 	addr := res.Address
 	conn, err := grpc.Dial(
 		addr,
-		grpc.WithTransportCredentials(r.creds),
+		grpc.WithUnaryInterceptor(r.clientCallInterceptor),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
 	)
 	if err != nil {
 		return nil, fmt.Errorf("error connecting to '%s' for namespace '%s'", addr, ns)
 	}
 	return routerv1.NewMsgClient(conn), nil
 }
-func loadClientCredentials(path string) (credentials.TransportCredentials, error) {
-	// Load certificate of the CA who signed server's certificate
-	pemServerCA, err := os.ReadFile(path)
-	if err != nil {
-		return nil, err
-	}
 
-	certPool := x509.NewCertPool()
-	if !certPool.AppendCertsFromPEM(pemServerCA) {
-		return nil, errors.New("failed to add server CA's certificate")
-	}
-
-	// Create the credentials and return it
-	config := &tls.Config{
-		RootCAs:    certPool,
-		MinVersion: tls.VersionTLS12,
-	}
+// intercepts grpc invocations and injects the router-routerKey.
+func (r *routerImpl) clientCallInterceptor(
+	ctx context.Context, method string, req, reply any, cc *grpc.ClientConn,
+	invoker grpc.UnaryInvoker, opts ...grpc.CallOption,
+) error {
+	md := metadata.New(map[string]string{"router-routerKey": r.routerKey})
+	ctx = metadata.NewOutgoingContext(ctx, md)
```

Comment:
This may be overly pedantic, but it would probably be safer to use:
```
ctx = metadata.AppendToOutgoingContext(ctx, "router-routerKey", r.routerKey)
```
To make sure you're not overwriting any existing metadata.

Reply:
whoops, that darn refactor tool!

Reply:
will address that as well

Reply:
client call interceptors have been removed for PerRPCCredential option.